client.py

```diff
@@ -4,17 +4,37 @@
 from graphics2 import gui
 
 
+class client:
+    def __init__(self):
+        self.scn = scenario([],[],[],[],[],[],[])
+        self.gui = gui(self.scn)
+        # Note: Dom dislikes the design where gui seems to own an object which is externally mutated.
 
-scn = scenario([],[],[],[],[],[],[])
-gui = gui(scn)
+    async def consume(self,ws):
+        try:
+            while True:
+                inbound = await ws.recv()
+                print("got msg from server.")
+                self.scn.decode(inbound)
+        except websockets.exceptions.ConnectionClosed:
+            pass
+        finally:
+            print("consume done")
 
-async def connect():
-    async with websockets.connect("ws://localhost:51010") as ws:
-        await ws.send("join")
-        scn.decode(await ws.recv())
-        
-        
-asyncio.get_event_loop().run_until_complete(connect())
+    async def produce(self,ws):
+        while True:
+            await asyncio.sleep(.001)
```

Comment:
Note: this was the crux of the issue.
https://stackoverflow.com/questions/37663560/websocket-recv-never-returns-inside-another-event-loop